Add button to copy code snippet

ryonakano opened the pull request:
Fixes #5
Closes #31

Differences from the previous PR:

- Use a label instead of the `edit-copy` icon, so that we can also use it to tell users the text being copied
- Do not swallow Ctrl + C in case someone selects some part of the snippet and press that hotkey
  - Now we require GLib >= 2.78 because of `Timeout.add_once ()`

I tried the button to be similar to the one written in the Prior Art of #5, but I'm not familiar with CSS so design feedback is especially appreciated.

![スクリーンショット 2024-08-14 17 44 58](https://github.com/user-attachments/assets/844f567a-7969-44c9-b0d1-61c1c4434389)

zeebok commented:
I won't merge in case there is design feedback, but the code looks good

danirabbit commented on data/Application.css:
```diff
@@ -40,3 +40,9 @@
     padding: 1.33rem;
 }
+
+.copy-button {
+    background: none;
+    border: 1px solid @borders;
+    box-shadow: none;
+}
```
I don't think we need to add this style because it actually makes the button much harder to read than our normal button style

danirabbit commented on src/IconView.vala:
```diff
@@ -103,6 +104,26 @@ public class IconView : Gtk.Box {
+        copy_button.add_css_class ("copy-button");
+
+        var copy_visible_controller = new Gtk.EventControllerMotion ();
+        copy_visible_controller.bind_property ("contains-pointer", copy_button, "visible",
+                                                BindingFlags.DEFAULT | BindingFlags.SYNC_CREATE);
```
Only showing on hover is problematic because it means this feature is unavailable on touch. It looks like on the docs site the button also shows if the code snippet contains focus. So we could do that or just always show it

This is kind of a funky indentation style. If you don't include the `BindingFlags` namespace you can have all the args on one line like:
```suggestion
        copy_visible_controller.bind_property (
            "contains-pointer", copy_button, "visible", DEFAULT | SYNC_CREATE
        );
```

danirabbit commented on src/IconView.vala:
```diff
@@ -103,6 +104,26 @@ public class IconView : Gtk.Box {
+        copy_button = new Gtk.Button.with_label (_("Copy")) {
+            valign = START,
+            halign = END,
+            visible = false,
```
Instead of using `visible` property, we can add a Gtk.Revealer to get a nice fade in animation. Just make sure you set `overflow` visible on the revealer to not clip shadows

danirabbit commented on src/IconView.vala:
```diff
@@ -103,6 +104,26 @@ public class IconView : Gtk.Box {
+            valign = START,
+            halign = END,
+            visible = false,
+            margin_top = 8,
+            margin_end = 8
```
Margins should be multiples of 3 :)

```suggestion
            margin_top = 6,
            margin_end = 6
```

ryonakano commented:
@danirabbit Thank you for your detailed reviews! I addressed your comments.

I don't have PCs with a touch screen so I can't test if the copy button get visible on these devices. Could you test that if you have one?

danirabbit reviewed (approved):
Yup that works for me! Can access the button with mouse, keyboard, and touch. Nice job :)
